PowerValidatedSolutions.psm1

```diff
@@ -48145,6 +48145,383 @@ Function Test-WMSubnetInput {
+            $null = Invoke-WebRequest -Uri "https://$caFqdn/certsrv/certnew.p7b?ReqID=CACert&Enc=b64" -Credential $domainCreds -OutFile $rootCaFilePathTemp -SkipCertificateCheck
+            openssl pkcs7 -inform PEM -outform PEM -in $rootCaFilePathTemp -print_certs > $rootCaFilePath
+        }
+    } catch {
+        Debug-ExceptionWriter -object $_
+    }
+
+    # Clean up
+    Remove-Item $rootCaFilePathTemp
+
+}
+Export-ModuleMember -Function Invoke-RequestSignedCertificateByMsca
+
+Function Invoke-GenerateChainPemFile {
+<#
+        .SYNOPSIS
+        Combine private key, signed certificate and/or root certificate chain into a single PEM file.   
+
+        .DESCRIPTION
+        The Invoke-GenerateChainPemFile cmdlet takes in private key, signed certificate and/or root certificate files and combine them into a single PEM file.  
+        
+        .EXAMPLE
+        Invoke-GenerateChainPemFile -certDir "C:\certificates" -keyFilePath "C:\certificates\xint-vrli01.rainpole.io.key" -crtFilePath 
+        "C:\certificates\xint-vrli01.rainpole.io.crt" -rootCaFilePath "C:\certificates\rpl-ad01.rainpole.io-rootCA.pem"
+        This example will combine xint-vrli01's private key, signed certificate and rpl0-ad01's root chain certificate into single
+        xint-vrli01.bundle.pem file. 
+
+        .EXAMPLE
+        Invoke-GenerateChainPemFile -certDir "C:\certificates" -crtFilePath "C:\certificates\xint-vrli01.rainpole.io.crt" 
+        -rootCaFilePath "C:\certificates\rpl-ad01.rainpole.io-rootCA.pem"
+        This example will combine xint-vrli01's signed certificate and rpl0-ad01's root chain certificate into single
+        xint-vrli01.bundle.pem file. 
+
+        .EXAMPLE
+        Invoke-GenerateChainPemFile -certDir "C:\certificates" -keyFilePath "C:\certificates\xint-vrli01.rainpole.io.key" -crtFilePath 
+        "C:\certificates\xint-vrli01.rainpole.io.crt"
+        This example will combine xint-vrli01's private key and signed certificate into single xint-vrli01.bundle.pem file. 
+
+        .PARAMETER certDir
+        The directory path to store the combined certificate bundle file.
+
+        .PARAMETER keyFilePath
+        The full file path for the private key file. 
+
+        .PARAMETER crtFilePath
+        The full file path for the signed certificate file. 
+
+        .PARAMETER rootCaFilePath
+        The full file path for the Certificate Authority root certificate file. 
+    #>
+
+    Param (
+        [Parameter (Mandatory = $true)] [ValidateNotNullOrEmpty()] [String] $certDir,
+        [Parameter (Mandatory = $false)] [ValidateNotNullOrEmpty()] [String] $keyFilePath,
+        [Parameter (Mandatory = $true)] [ValidateNotNullOrEmpty()] [String] $crtFilePath,            
+        [Parameter (Mandatory = $false)] [ValidateNotNullOrEmpty()] [String] $rootCaFilePath
+    )
+
+    # Validate inputs
+    if (-Not (Test-Path -Path $certDir)) {
+        Write-Error "Error: $certDir directory does not exist." -ErrorAction Stop
```

Comment:
```suggestion
        Write-Error "Certificate directory ($certDir) does not exist: PRE_VALIDATION_FAILED" -ErrorAction Stop
```

Reply:
Addressed in commit 53039acbaaffafaf219274cac1a3684735130f53